src/FileExtensionManager.cpp

```diff
@@ -0,0 +1,108 @@
+#include "FileExtensionManager.h"
+#include "ui_FileExtensionManager.h"
+
+#include "FileDialog.h"
+
+FileExtensionManager::FileExtensionManager(QStringList init, QWidget *parent) :
+    QDialog(parent),
+    ui(new Ui::FileExtensionManager)
+{
+    ui->setupUi(this);
+
+    int i = 0;
+    foreach(QString itemString, init)
+    {
+        ui->tableExtensions->insertRow(i);
+        QString description = itemString.left(itemString.indexOf('(')).trimmed();
+        QString extension   = itemString;
+        extension = extension.remove (0, itemString.indexOf('(')+1).remove(')').simplified().trimmed();
+        QTableWidgetItem *newItemDescription = new QTableWidgetItem(description);
+        QTableWidgetItem *newItemExtension   = new QTableWidgetItem(extension);
+        ui->tableExtensions->setItem(i, 0, newItemDescription);
+        ui->tableExtensions->setItem(i, 1, newItemExtension);
+        i++;
+    }
+
+    connect(ui->buttonAdd, SIGNAL(clicked(bool)), this, SLOT(addItem()));
+    connect(ui->buttonRemove, SIGNAL(clicked(bool)), this, SLOT(removeItem()));
+
+    connect(ui->buttonDown, SIGNAL(clicked(bool)), this, SLOT(downItem()));
+    connect(ui->buttonUp, SIGNAL(clicked(bool)), this, SLOT(upItem()));
+}
+
+FileExtensionManager::~FileExtensionManager()
+{
+    delete ui;
+}
+
+void FileExtensionManager::addItem()
+{
+    int i = ui->tableExtensions->rowCount();
+    ui->tableExtensions->insertRow(i);
+    QTableWidgetItem *newItemDescription = new QTableWidgetItem("Description");
+    QTableWidgetItem *newItemExtension   = new QTableWidgetItem("*.extension");
```

Comment:
For these two lines you should add ```tr(...)``` around the user-visible strings to make them translatable.